package main

import (
	"errors"
	"fmt"
)

func returnError(a, b int) error {
	if a == b {
		err := errors.New("error generated by function")
		return err
	} else {
		return nil
	}
}

func main() {
	err := returnError(1, 2)
	if err == nil {
		fmt.Println("function ended normally")
	} else {
		fmt.Println(err)
	}

	err = returnError(2, 2)
	if err == nil {
		fmt.Println("function ended normally")
	} else {
		fmt.Println(err)
	}

	if err.Error() == "error generated by function" {
		fmt.Println("!!")
	}
}
